C++/syntax_test_cpp.cpp

```diff
@@ -2547,6 +2547,23 @@ MyEnum MACRO1
 // Test preprocessor branching and C blocks
 /////////////////////////////////////////////
 
+int bar(int, int const *, int const * const);
+/*  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.function */
+/*     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.function.parameters meta.group */
+/*     ^ punctuation.section.group.begin */
+/*                                         ^ punctuation.section.group.end */
+/*                                          ^ punctuation.terminator */
+/*      ^^^ storage.type */
+/*         ^ punctuation.separator */
+/*           ^^^ storage.type */
+/*               ^^^^^ storage.modifier */
+/*                     ^ keyword.operator */
+/*                      ^ punctuation.separator */
+/*                        ^^^ storage.type */
+/*                            ^^^^^ storage.modifier */
+/*                                  ^ keyword.operator */
+/*                                    ^^^^^ storage.modifier */
```

Comment:
Is this intentional or a mis-highlight?

Reply:
> Is this intentional or a mis-highlight?

Is what intentional?

Reply:
Yes. `const` can have multiple layers when used in C. `const` can also be applied to the layers of pointers that a type can have. Hence why it can occur here. Otherwise, only part of the type will be `const`. And `const` usually only applies to whatever immediately precedes, except for the special case where `const` comes before the type. For example, `const char` is equivalent to `char const`. If this was a pointer, the array elements would be `const` but not the pointer itself.